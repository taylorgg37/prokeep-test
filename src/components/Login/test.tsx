import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'

import { Login } from '.'

describe('<Login />', () => {
  it('The form should be visible', () => {
    const { container } = render(<Login />)

    expect(screen.getByText('Email')).toBeInTheDocument()
    expect(screen.getByText('Password')).toBeInTheDocument()
    expect(screen.getByText('Login')).toBeInTheDocument()

    expect(container).toMatchSnapshot()
  })
  it('Submit the form with a fake email address and password', async () => {
    const { container } = render(<Login />)
    const user = userEvent.setup()
    const emailInput = container.querySelector('#email')
    const PasswordInput = container.querySelector('#password')
    const SubmitBtn = container.querySelector('#submit-btn')
    if (emailInput) {
      await user.type(emailInput, 'test@mail.com')
    }

    if (PasswordInput) {
      await user.type(PasswordInput, 'password123')
    }

    if (SubmitBtn) {
      await user.click(SubmitBtn)
    }
  })
  it('Submit the form with an empty email address and password', async () => {
    const { container } = render(<Login />)
    const user = userEvent.setup()
    const SubmitBtn = container.querySelector('#submit-btn')
    expect(SubmitBtn).toBeVisible()

    await user.click(SubmitBtn as Element)
    expect(screen.getByText('Enter your email.')).toBeVisible()
    expect(screen.getByText('Enter your password.')).toBeVisible()
  })
})
