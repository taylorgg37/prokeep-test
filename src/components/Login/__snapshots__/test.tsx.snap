// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Login /> > Should be visible form 1`] = `
<div>
  <div
    class="flex min-h-screen flex-col items-center justify-center bg-gradient-to-br from-red-600 to-pink-500"
  >
    <div
      class="max-w-md rounded-lg bg-white p-8 shadow-lg"
    >
      <h1
        class="mb-8 text-center text-4xl font-bold text-purple-700"
      >
        Welcome to My Website
      </h1>
      <form
        class="space-y-6"
      >
        <div>
          <label
            class="mb-2 block font-bold text-gray-700"
            for="email"
          >
            Email
          </label>
          <input
            class="w-full rounded-lg border border-gray-400 px-4 py-2"
            id="email"
            name="email"
            placeholder="name@example.com"
            type="email"
          />
        </div>
        <div>
          <label
            class="mb-2 block font-bold text-gray-700"
            for="password"
          >
            Password
          </label>
          <input
            class="w-full rounded-lg border border-gray-400 px-4 py-2"
            id="password"
            name="password"
            placeholder="qwerty123"
            type="password"
          />
        </div>
        <div>
          <button
            class="w-full rounded-lg bg-purple-700 px-4 py-2 font-bold text-white hover:bg-purple-900"
            id="submit-btn"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;

exports[`<Login /> > The form should be visible 1`] = `
<div>
  <div
    class="flex min-h-screen flex-col items-center justify-center bg-gradient-to-br from-red-600 to-pink-500"
  >
    <div
      class="max-w-md rounded-lg bg-white p-8 shadow-lg"
    >
      <h1
        class="mb-8 text-center text-4xl font-bold text-purple-700"
      >
        Welcome to My Website
      </h1>
      <form
        class="space-y-6"
      >
        <div>
          <label
            class="mb-2 block font-bold text-gray-700"
            for="email"
          >
            Email
          </label>
          <input
            class="w-full rounded-lg border border-gray-400 px-4 py-2"
            id="email"
            name="email"
            placeholder="name@example.com"
            type="email"
          />
        </div>
        <div>
          <label
            class="mb-2 block font-bold text-gray-700"
            for="password"
          >
            Password
          </label>
          <input
            class="w-full rounded-lg border border-gray-400 px-4 py-2"
            id="password"
            name="password"
            placeholder="qwerty123"
            type="password"
          />
        </div>
        <div>
          <button
            class="w-full rounded-lg bg-purple-700 px-4 py-2 font-bold text-white hover:bg-purple-900"
            id="submit-btn"
          >
            Login
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
